quipucords/fingerprinter/jboss_fuse.py

```diff
@@ -55,12 +84,25 @@ def detect_jboss_fuse(source, facts):
     fuse_on_karaf = facts.get(JBOSS_FUSE_ON_KARAF_KARAF_HOME)
     systemctl_files = facts.get(JBOSS_FUSE_SYSTEMCTL_FILES)
     chkconfig = facts.get(JBOSS_FUSE_CHKCONFIG)
-    fuse_activemq = facts.get(JBOSS_ACTIVEMQ_VER, [])
-    fuse_camel = facts.get(JBOSS_CAMEL_VER, [])
-    fuse_cxf = facts.get(JBOSS_CXF_VER, [])
     subman_consumed = facts.get(SUBMAN_CONSUMED, [])
     entitlements = facts.get(ENTITLEMENTS, [])
-    fuse_versions = list(set(fuse_activemq + fuse_camel + fuse_cxf))
+    # Get versions from regular scan
+    fuse_activemq = facts.get(JBOSS_FUSE_ACTIVEMQ_VER, [])
+    fuse_camel = facts.get(JBOSS_FUSE_CAMEL_VER, [])
+    fuse_cxf = facts.get(JBOSS_FUSE_CXF_VER, [])
+    # Get fuse on eap versions from a regular scan
+    eap_camel = facts.get(JBOSS_FUSE_ON_EAP_CAMEL_VER, [])
+    eap_activemq = facts.get(JBOSS_FUSE_ON_EAP_ACTIVEMQ_VER, [])
+    eap_cxf = facts.get(JBOSS_FUSE_ON_EAP_CXF_VER, [])
+    eap_versions = get_versions([eap_camel, eap_activemq, eap_cxf])
```

Comment:
I think you can simplify this by changing to `get_versions(eap_camel + eap_activemq + eap_cxf)` and updating get_versions to expect a single list instead of a nested one. (My version is potentially slower than yours, but I don't expect this to be performance-sensitive.)